presto/utils.py

```diff
@@ -314,6 +301,7 @@ def process_parquet(df: pd.DataFrame) -> pd.DataFrame:
         df = pd.concat([df, dummy_df])
 
     # finally pivot the dataframe
+    index_columns = list(np.unique(index_columns))
     df_pivot = df.pivot(index=index_columns, columns="timestamp_ind", values=feature_columns)
     df_pivot = df_pivot.fillna(NODATAVALUE)
```

Comment:
this is the actual filling of timestamps "in the middle" ?

Reply:
this part is now actually obselete, because ranking function that is used to create timestamp_ind does not make any gaps that might require filling in. I removed it for now b4964b04312bfbc47f4286590a539cabaf014475